proposals/esm-integration/README.md

```diff
@@ -29,121 +29,112 @@ WebAssembly
 A declarative API would improve the ergonomics by making this work happen implicitly.
 
 ```js
-import {foo} from "./myModule.wasm";
+import { foo } from "./myModule.wasm";
 foo();
 ```
 
-### Enable WebAssembly modules to take part in JavaScript module graphs
-
-With the introduction of ES modules, the JS spec provides a way to express module dependency graphs.
-
-These graphs are expressed with import and export statements.
+Then by integrating with [Source Phase Imports](https://github.com/tc39/proposal-source-phase-imports), arbitrary instantiations with custom imports can still be supported.
 
 ```js
-// main.js
-
-import {count, incrementCount} from "./dep.js"
+import source myModule from "./myModule.wasm");
+const { foo: foo1 } = await WebAssembly.instantiate(myModule, { ...imports });
```

Comment:
This would also work with sync instantiate (`new WebAssembly.Instance(...)`), which would be symmetric with the current proposal. It might be worth having such an example here given that avoiding top-level await seems to be a hot topic in JS right now.

Reply:
Good point, I've updated one of the examples to use synchronous instantiation, while retaining async instantiation in the one that was already using top-level await.